### PR TITLE
refactor: Migrate App to Typescript

### PR DESCRIPTION
**Pull request by thisisamir98**

### CODE REVIEW

File: src/script/main/app.ts [hunk 1]
```diff
@@ -120,12 +121,18 @@ import {ConnectionService} from '../connection/ConnectionService';
 import {TeamService} from '../team/TeamService';
 import {SearchService} from '../search/SearchService';
 import {CryptographyService} from '../cryptography/CryptographyService';
-import {AccessTokenError} from '../error/AccessTokenError';
+import {AccessTokenError, ACCESS_TOKEN_ERROR_TYPE} from '../error/AccessTokenError';
 import {ClientError} from '../error/ClientError';
 import {AuthError} from '../error/AuthError';
 import {AssetRepository} from '../assets/AssetRepository';
+import {SQLeetEngine} from '@wireapp/store-engine-sqleet';
```
Comment (ffflorian) [inline src/script/main/app.ts]: Please move this import to the top, next to the other external imports. Also please use `import type`.

File: src/script/main/app.ts [hunk 2]
```diff
@@ -138,6 +145,25 @@ function doRedirect(signOutReason) {
 }
 
 class App {
+  apiClient: APIClient;
+  backendClient: BackendClient;
+  logger: any;
```
Comment (ffflorian) [inline src/script/main/app.ts]: Please import `Logger` and use it here.

File: src/script/main/app.ts [hunk 3]
```diff
@@ -941,14 +945,14 @@ $(async () => {
       restUrl: Config.getConfig().BACKEND_REST,
       webSocketUrl: Config.getConfig().BACKEND_WS,
     });
-    const shouldPersist = loadValue(StorageKey.AUTH.PERSIST);
+    const shouldPersist: boolean = loadValue(StorageKey.AUTH.PERSIST);
```
Comment (ffflorian) [inline src/script/main/app.ts]:
```suggestion
    const shouldPersist = loadValue<boolean>(StorageKey.AUTH.PERSIST);
```

File: src/script/main/app.ts [hunk 4]
```diff
@@ -891,32 +921,6 @@ class App {
...
-  }
-
-  /**
-   * Report call telemetry to Raygun for analysis.
-   * @returns {undefined} No return value
-   */
-  reportCall() {
-    this.repository.calling.reportCall();
-  }
-
```
Comment (ffflorian) [inline src/script/main/app.ts:919]: What about this removed code? Don't we need it anymore?

File: src/script/main/app.ts [hunk 5]
```diff
@@ -388,7 +418,7 @@ class App {
         properties: propertiesRepository,
         team: teamRepository,
         user: userRepository,
-      } = this.repository;
```
Comment (ffflorian) [inline src/script/main/app.ts]: The type declaration shouldn't be needed since `this.repository` is typed.

**Review by ffflorian — CHANGES_REQUESTED** [review]

File: src/script/main/app.ts (hunk 4 above)
Comment (thisisamir98) [reply src/script/main/app.ts:919]: this is not used anywhere at all, calling repo has no method called `reportCall`

File: src/script/main/app.ts (hunk 5 above)
Comment (thisisamir98) [reply src/script/main/app.ts]: done in https://github.com/wireapp/wire-webapp/pull/9511/commits/f89353a3a5793eaa4bf6a95c1cd629ec63828ee1

File: src/script/main/app.ts (hunk 2 above)
Comment (thisisamir98) [reply src/script/main/app.ts]: done in https://github.com/wireapp/wire-webapp/pull/9511/commits/f89353a3a5793eaa4bf6a95c1cd629ec63828ee1

File: src/script/main/app.ts (hunk 1 above)
Comment (thisisamir98) [reply src/script/main/app.ts]: done in https://github.com/wireapp/wire-webapp/pull/9511/commits/f89353a3a5793eaa4bf6a95c1cd629ec63828ee1

**Review by ffflorian — APPROVED** [review]

**Pull request merged**